### DISCUSSION TITLE
NTLM or Kerberos authentication in HTTP listener?

### DISCUSSION THREAD

**Discussion by NoraTheExplorer** [nextgenhealthcare/connect#5526] [Q&A]
Hi,

is it possible to get this working? As far as I understand I have to do the following:

- check the first request header for the "authentication" key
- if not present send a 401.2 message back including the WWW-authenticate header
- if the client closes the connection and sends the clients user name, computer name and domain name in a new request as an encoded string to the server
- this information is forwarded to the domain controller to create a challenge
- this challenge is sent back to the client in another 401 (401.1) message
- the client initiates a new request and creates a hash out of the challenge and it's own user password and sends it again back to the server
- the server sends the hash to the domain controller to validate it
- if everything is correct, access is granted

So, now my question is, can this be achieved with MirthConnect integrated tools or do I have to script it on my own?

**Comment by siddharthsharma9258** [comment]
Sounds complicated. Don't know if it can be pulled off in Mirth.

**Comment by kirbykn2** [comment]
I think yes!

On Fri, Nov 11, 2022 at 2:36 PM Siddharth Sharma ***@***.***>
wrote:

> Sounds complicated. Don't know if it can be pulled off in Mirth.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5526#discussioncomment-4120115>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWDY4SMFBFJBX6WYWO3LWH2N2VANCNFSM6AAAAAAR5Z2EDQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <nextgenhealthcare/connect/repo-discussions/5526/comments/4120115@
> github.com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Comment by pacmano1** [comment]
It's certainly possible, but you need to code most of that.
